### PR TITLE
Create specs for customer service dashboard

### PR DESCRIPTION
**Pull request by marionf**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Create specs for customer service dashboard

### CODE REVIEW

File: back-office/customer-service/customer-service/dashboard.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+### Bulk actions drop-down
+
+This dropdown is enabled only if at least one message is selected, otherwise it's disabled.
+
+There is 5 options:
+
+1) Mark as "opened"
+
+After clicking on it, the status of the selected messages are changed for "opened" and a successfull message is displayed "The status of the selection has been successfully changed."
```
Comment (marionf) [inline back-office/customer-service/customer-service/dashboard.md:79]: @Julievrz this successfull message is new, could you please verify the it ? :)

**Comment by marionf** [comment]
@prestascott this the specs for the customer service page (just the dashboard, I will make the specs for the view page later), if you want to take a look ;)

File: back-office/customer-service/customer-service/dashboard.md [hunk 2]
```diff
@@ -267,38 +267,38 @@ A tooltip is available: "User to use to connect to your IMAP server."
 
 A tooltip is available: "Password to user to connecto to your IMAP server."
```
Comment (Julievrz) [inline back-office/customer-service/customer-service/dashboard.md:268]:
```suggestion
A tooltip is available: "Password to use to connect to your IMAP server."
```

File: back-office/customer-service/customer-service/dashboard.md (hunk 1 above)
Comment (Julievrz) [reply back-office/customer-service/customer-service/dashboard.md:79]: Hi @marionf, there is a string that already exists: "The status has been successfully updated." in Admin.Notifications.Success 
Couldn't we keep this one? Or is this wording going to apply only when several messages are selected?

Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:79]: This message will be displayed only when several messages are selected.
Maybe we can simply use the existing wording: "Successful update.", wdyt ?

Comment (Julievrz) [reply back-office/customer-service/customer-service/dashboard.md:79]: I think we can use a more general wording, yes. We tend to prefer "Update successful" 😊 (without punctuation)

Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:79]: Ok, so let's go for this one, thank you @Julievrz !

**Comment by marionf** [comment]
@matks As this page will be soon migrated, maybe you want to take a look to the specs, or I let you ping the developer that will be working on it ;)

File: back-office/customer-service/customer-service/dashboard.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+After clicking on it, the thread is opened in the same tab.
+This action can also be performed by clicking on a line.
+
+- Delete, the second one:
+
+After clicking on it, a modal is opened asking to confirm or cancel the action: "Are you sure you want to delete the selected item(s)?"
+
+The action can be canceled by clicking on the cross or on the cancel button.
+
+After clicking on "Delete" button, the message is deleted and a successfull message is displayed "Successful deletion."
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:220]:
```suggestion
After clicking on "Delete" button, the message is deleted and a successful message is displayed "Successful deletion."
```

File: back-office/customer-service/customer-service/dashboard.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+- Group 2
+  - Shop C
+  
+If I select group 1 as context, then the messages created in Shop A and Shop B will be displayed.
+
+If I select Shop A as context, then the messages created in Shop A will be displayed.
+
+If I select all shops as context, then the messages created in Shop A, Shop B and Shop C will be displayed.
+
+#### Select all / unselect all checkbox
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:133]: Sorry, I can't find it on the mockup: is it two items of the bulk action dropdown, or do those buttons appear in some other place?

File: back-office/customer-service/customer-service/dashboard.md [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+#### Pagination
+
+The pagination of the list starts when there are more than 20 messages.
+
+The merchant can select the number of messages displayed through a drop-down: 20, 50, 300, 1000
+
+When there are more results than the number selected, the merchant can navigate through pages.
+
+If a sort and/or a search have been done, they should be keeped after changing page.
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:244]:
```suggestion
If a sort and/or a search have been done, results should be kept when moving on to the next page.
```

File: back-office/customer-service/customer-service/dashboard.md [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+
+## CONTACT OPTIONS
+
+The user can edit 2 options, which apply to all contacts:
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:248]: Or 'customers'? I'm getting confused! 😄

File: back-office/customer-service/customer-service/dashboard.md [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+
+1) **Allow file uploading**: This toggle allows the merchant to choose if he/she wants to allow his/her visitors to upload files through the contact form.
+
+It's enabled by default.
+
+When enabled, an optional field "Attachment" is displayed in the front-office's contact form and a "choose file" CTA allows to select a file.
+
+A tooltip is available: "Allow customers to upload files using the contact page."
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:256]: Do you have the list of the accepted extensions? Just to test, I've tried to upload a .xls file and I got an error notification: `Bad file extension`.

File: back-office/customer-service/customer-service/dashboard.md [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+
+2) **Default message**: This field allows the merchant to enter a default message that will be pre-filled while responding to a message.
+
+By default this field is filled with the following message:
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:260]: What an alliteration! 😎

File: back-office/customer-service/customer-service/dashboard.md [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+By default this field is filled with the following message: 
+
+"Dear Customer,
+
+Regards,
+Customer service"
+
+A tooltip is available: "Please fill out the message fields that appear by default when you answer a thread on the customer service page"
+
+A drop-down next to the field is displayed when there are several languages installed (disabled languages are diplayed in the drop-down). It allows to choose in which language the field is displayed.
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:269]:
```suggestion
A drop-down next to the field is displayed when there are several languages installed (disabled languages are displayed in the drop-down). It allows choosing in which language the field is displayed and customizing the message according to the customer's language.
```

File: back-office/customer-service/customer-service/dashboard.md [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+
+A tooltip is available: "Do not user start-TLS to encrypt the session, even with servers that support it."
+
+### Multistore
+
+In a multistore context, the options listed above have [checkboxes](https://github.com/PrestaShop/PrestaShop/issues/19378) and [specific settings drop-down](https://github.com/PrestaShop/PrestaShop/issues/19330), take a look at the conditions and specs [here](https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/multistoregeneralspecs.md)
+
+### SAVE BUTTON
+
+When clicking on save, if all settings have well been saved, then a successfull message is displayed: "The settings have been successfully updated."
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:340]:
```suggestion
When clicking on save, if all settings have well been saved, then a successful message is displayed: "The settings have been successfully updated."
```

File: back-office/customer-service/customer-service/dashboard.md [hunk 11]
```diff
@@ -0,0 +1,354 @@
...
+
+If all IMAP fields are not filled and if you click on save, then the following information message is displayed: "IMAP configuration is not correct"
```
Comment (LouiseBonnard) [inline back-office/customer-service/customer-service/dashboard.md:342]: Is it blocking to save the configuration? Should it be a warning/error notification instead?

File: back-office/customer-service/customer-service/dashboard.md [hunk 12]
```diff
@@ -0,0 +1,354 @@
...
+https://www.figma.com/proto/axasD3Kyg52okWHqvWPN1I/PrestaShop---Customer-Service-(migration)?node-id=12%3A5267&scaling=min-zoom
+https://www.figma.com/proto/axasD3Kyg52okWHqvWPN1I/PrestaShop---Customer-Service-(migration)?node-id=12%3A9835&scaling=min-zoom
+https://www.figma.com/proto/axasD3Kyg52okWHqvWPN1I/PrestaShop---Customer-Service-(migration)?node-id=12%3A8042&scaling=min-zoom
+https://www.figma.com/proto/axasD3Kyg52okWHqvWPN1I/PrestaShop---Customer-Service-(migration)?node-id=12%3A11958&scaling=min-zoom
+
+## MESSAGES NOTIFICATIONS
+
+When a new messages is sent, then it's displayed in the notifications in the header.
+
+If you click on it, the thread is opened (view page) and all the messages notifications are removed.
```
[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md]: No, I am talking of the bell in the header

capture d'écran_2914

File: back-office/customer-service/customer-service/dashboard.md [hunk 13]
```diff
@@ -0,0 +1,354 @@
...
+
+2) Export to SQL Manager
+
+After clicking on it, the SQL query is opened in Advanced parameters > Database > SQL Manage
+
+### Bulk actions drop-down
+
+This dropdown is enabled only if at least one message is selected, otherwise it's disabled.
+
+There is 5 options, see the improvement [#19197](https://github.com/PrestaShop/PrestaShop/issues/19197)
```
[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:88]: It seems a good idea to let the user change the status from the listing for a single line (and also in bulk)
@prestascott what do you think ?

File: back-office/customer-service/customer-service/dashboard.md [hunk 14]
```diff
@@ -0,0 +1,354 @@
...
+
+4) Type
+
+The type of the messsage selected by the customer in the contact form is displayed.
+
+It's possible to sort by type asc or desc and search by selecting the type through the drop-down, where all contacts' title are displayed.
+
+5) Language
+
+The language selected by the customer in front-office while sending the message is displayed.
```
[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:170]: I just tried, it seems it's only the language used by the customer in the front office

[the remark this answers is not in the document]
Comment (LouiseBonnard) [reply back-office/customer-service/customer-service/dashboard.md:170]: I also tried yesterday, with PrestaShop 1.7.7, and when I changed the language in the back office, wrote an order message, and then went to the Customer Service page, I could see that the language displayed in this column had changed.

File: back-office/customer-service/customer-service/dashboard.md (hunk 4 above)
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:139]: I am talking of this checkbox allowing to select all / unselect all messages

capture d'écran_2915

File: back-office/customer-service/customer-service/dashboard.md [hunk 15]
```diff
@@ -0,0 +1,354 @@
...
+It's possible to sort by customer asc or desc and search by customer (firstname and/or lastname)
+
+3) Email
+
+The customer's email address is displayed.
+It's possible to sort by email asc or desc and search by email
+
+4) Type
+
+The type of the messsage selected by the customer in the contact form is displayed.
```
[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:164]: It's the contact type (customer service, webmaster ...)

File: back-office/customer-service/customer-service/dashboard.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:170]: Ahh, yes through the order page, you are right !

File: back-office/customer-service/customer-service/dashboard.md [hunk 16]
```diff
@@ -0,0 +1,354 @@
...
+
+8) Messages
+
+The customer's message is displayed and truncated if it contains more than 40 chars.
+
+It's possible to sort by message asc or desc and search by messages
+
+9) Private
+
+The customer's message is displayed and truncated if it contains more than 40 chars.
```
[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md]: Damn ! Fixed, thanks :)

File: back-office/customer-service/customer-service/dashboard.md [hunk 17]
```diff
@@ -0,0 +1,354 @@
...
+
+There are 12 columns:
+
+1) ID
+
+It's possible to sort by id asc or desc and search by id.
+
+2) Customer
+
+The firstname and lastname are displayed if the customer was logged in while sending the message. Otherwise, "--" is displayed.
```
[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:153]: Yes, thank you I just added it :)

File: back-office/customer-service/customer-service/dashboard.md [hunk 18]
```diff
@@ -0,0 +1,354 @@
...
+
+The first letter of the employee's firstname and his/her lastname are displayed if the message is assigned to an employee. Otherwise, "--" is displayed.
+
+It's possible to sort by employee asc or desc and search by employee's lastname.
+
+8) Messages
+
+The customer's message is displayed and truncated if it contains more than 40 chars.
+
+It's possible to sort by message asc or desc and search by messages
```
[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:192]: There is this behavior on a lot of BO pages (legacy and symfony), example below with the order page, where you can search by a customer and also sort the results in an increasing or decreasing way
But the arrows to sort are displayed only on hover, unlike legacy pages

capture d'écran_2916

File: back-office/customer-service/customer-service/dashboard.md (hunk 8 above)
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:268]: héhé :star_struck:

File: back-office/customer-service/customer-service/dashboard.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (LouiseBonnard) [reply back-office/customer-service/customer-service/dashboard.md:192]: Absolutely! But this behavior is applied to specific columns only, not to all the columns of a table like in the legacy pages, that's why I was asking about it. In your example, only the _ID_ and the _Customer_ columns can be sorted by ascending or descending order.

File: back-office/customer-service/customer-service/dashboard.md [hunk 19]
```diff
@@ -0,0 +1,362 @@
...
+
+## CONTACT OPTIONS
+
+The user can edit 2 options, which apply to all contacts:
+
+1) **Allow file uploading**: This toggle allows the merchant to choose if he/she wants to allow his/her visitors to upload files through the contact form.
+
+It's enabled by default.
+
+When enabled, an optional field "Attachment" is displayed in the front-office's contact form and a "choose file" CTA allows to select a file.
```
Comment (marionf) [inline back-office/customer-service/customer-service/dashboard.md:262]:
```suggestion
When enabled, an optional field "Attachment" is displayed in the front-office's contact form and a "choose file" CTA allows to select a file.

Only the following extension are accepted: pdf, txt, png, jpg, gif, zip.
Otherwise, an error notfication is displayed: "Bad file extension".
```

File: back-office/customer-service/customer-service/dashboard.md (hunk 7 above)
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:264]: Yes, I just added them :)

File: back-office/customer-service/customer-service/dashboard.md (hunk 4 above)
Comment (LouiseBonnard) [reply back-office/customer-service/customer-service/dashboard.md:139]: Oh, that's why it was not listed as part of the bulk actions drop-down. I was wondering why it was not listed above instead. :-)

File: back-office/customer-service/customer-service/dashboard.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (LouiseBonnard) [reply back-office/customer-service/customer-service/dashboard.md:192]: My bad, the behavior stays the same, thanks @marionf!

File: back-office/customer-service/customer-service/dashboard.md (hunk 11 above)
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:353]: It's not blocking to save the configuration, a message "The settings have been successfully updated." is displayed but the user is warned that the configuration is not correct

File: back-office/customer-service/customer-service/dashboard.md [hunk 20]
```diff
@@ -0,0 +1,365 @@
...
+
+The merchant can select the number of messages displayed through a drop-down: 20, 50, 300, 1000
+
+When there are more results than the number selected, the merchant can navigate through pages.
+
+If a sort and/or a search have been done, results should be kept when moving on to the next page.
+
+## CONTACT OPTIONS
+
+The user can edit 2 options, which apply to all contacts:
```
Comment (marionf) [inline back-office/customer-service/customer-service/dashboard.md:256]:
```suggestion
The user can edit 2 options, which apply to all customers / visitors:
```

**Comment by marionf** [comment]
@LouiseBonnard Thank you for the review, all your comments have been addressed

File: back-office/customer-service/customer-service/dashboard.md (hunk 11 above)
Comment (LouiseBonnard) [reply back-office/customer-service/customer-service/dashboard.md:358]: In other words, the user can save an incorrect IMAP configuration? I feel like this behavior could be improved but I don't know how problematic it can be. In my opinion, we should turn this information notification into a warning one, or at least rephrase it to emphasize the fact that the configuration saved must not be left invalid.

Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:358]: Yes, the wording here is not really correct as it's not that the configuration is not correct but rather incomplete
Unless this message shows up in some other case that I haven't identified, but from what I see it shows until all IMAP fields are filled out.

Comment (LouiseBonnard) [reply back-office/customer-service/customer-service/dashboard.md:358]: I've looked into the documentation and it looks like the first 4 fields need to be filled to run the feature. So I suggest changing the behavior: IMAP URL, port, user, and password fields must be filled to be saved.

If the user tries to save an incomplete configuration, display an error notification: _Unable to save settings, please fill in all IMAP fields to configure your customer service options._ in _Admin.Catalog.Notification_.

If the user saves an incorrect configuration, display a warning notification (and not an information notification), cf. line 360 below.

@marionf, @Julievrz, what do you think?

Comment (Julievrz) [reply back-office/customer-service/customer-service/dashboard.md:358]: @LouiseBonnard Is imap not being installed on the server considered as an incorrect configuration?

Comment (LouiseBonnard) [reply back-office/customer-service/customer-service/dashboard.md:358]: Sorry for my late reply! From what I know, the user can set options to configure IMAP. It is not required though.

But if he/she starts configuring it but does not complete all the fields, it is not incorrect but incomplete (= it won't work). And if he/she fills all the fields in but the configuration is wrong, it is incorrect (= it still does not work).

File: back-office/customer-service/customer-service/dashboard.md [hunk 21]
```diff
@@ -0,0 +1,370 @@
...
+
+It's possible to sort by customer asc or desc and search by customer (firstname and/or lastname)
+
+3) Email
+
+The customer's email address is displayed.
+It's possible to sort by email asc or desc and search by email
+
+4) Type (contact)
+
```
Comment (Robin-Fischer-PS) [inline back-office/customer-service/customer-service/dashboard.md:170]: I don't know if it's useful to add this information, but when the contact is made via the "Add a message" in FO > User Account > Order History > Order, there is currently no "Type" of Contact in the listing (so "--" is displayed).

File: back-office/customer-service/customer-service/dashboard.md [hunk 22]
```diff
@@ -0,0 +1,370 @@
...
+https://www.figma.com/proto/axasD3Kyg52okWHqvWPN1I/PrestaShop---Customer-Service-(migration)?node-id=12%3A8042&scaling=min-zoom
+
+https://www.figma.com/proto/axasD3Kyg52okWHqvWPN1I/PrestaShop---Customer-Service-(migration)?node-id=12%3A11958&scaling=min-zoom
+
+## MESSAGES NOTIFICATIONS
+
+When a new message is sent by a customer or a visitor, then it's displayed in the notifications (bell) in the header.
+
+If you click on it, the thread is opened (view page) and all the messages notifications are removed.
```
Comment (Robin-Fischer-PS) [inline back-office/customer-service/customer-service/dashboard.md:19]: Maybe add the info that the notification is removed for this user only (but since it's the general behavior for notifications, it's might not be useful).

File: back-office/customer-service/customer-service/dashboard.md (hunk 21 above)
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:170]: Yes, good idea :+1: 
I added this here: https://github.com/PrestaShop/prestashop-specs/pull/187/commits/1c3f188a68e773b4f34de83849cce17b4e3c725b

File: back-office/customer-service/customer-service/dashboard.md (hunk 22 above)
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md]: Nice catch, thank you @Robin-Fischer-PS :)

File: back-office/customer-service/customer-service/dashboard.md (hunk 11 above)
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:360]: Yes, that's it :)

File: back-office/customer-service/customer-service/dashboard.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (prestascott) [reply back-office/customer-service/customer-service/dashboard.md:93]: I think it's a good improvement for the experience of the users and as @LouiseBonnard said, ux will stay consistent through our back office. 👍

[the remark this answers is not in the document]
Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md:93]: Added to the spec here: https://github.com/PrestaShop/prestashop-specs/pull/187/commits/8d4921d4bc0757cdb78ff3c7931d0b91795c9c9c

File: back-office/customer-service/customer-service/dashboard.md (hunk 11 above)
Comment (Julievrz) [reply back-office/customer-service/customer-service/dashboard.md:372]: I'm sorry but I don't understand which warning notification will be displayed in case of an incorrect configuration. 😅 
Is it "IMAP is not installed on this server"?

Comment (Julievrz) [reply back-office/customer-service/customer-service/dashboard.md:372]: "IMAP is not installed", but how should we install it? Is "installing" and "configuring" the same? I'm lost sorry.🤣🤔

Comment (Julievrz) [reply back-office/customer-service/customer-service/dashboard.md:372]: @LouiseBonnard What do you think of: 'IMAP could not be installed on the server.'

Comment (LouiseBonnard) [reply back-office/customer-service/customer-service/dashboard.md:372]: Very clear! I'd just add an element of solution to help users know what to do. For instance, _IMAP could not be installed on the server, check your configuration and try again._ or any other wording that could inform the merchants they need to check the information filled.

Comment (Julievrz) [reply back-office/customer-service/customer-service/dashboard.md:372]: Yes, sounds great!

Comment (marionf) [reply back-office/customer-service/customer-service/dashboard.md]: I updated the notifications here: https://github.com/PrestaShop/prestashop-specs/pull/187/commits/5ae453058a2427a7e25d8afdc239663edd275743
Is it ok for you @Julievrz ?

Comment (Julievrz) [reply back-office/customer-service/customer-service/dashboard.md]: Yes, perfect! thank you!

**Comment by marionf** [comment]
@LouiseBonnard @Robin-Fischer-PS 
Thank you for the review, I applied all your suggestions
I let you check one last time before the merge

**Review by Robin-Fischer-PS — APPROVED** [review]

**Review by LouiseBonnard — APPROVED** [review]

**Comment by LouiseBonnard** [comment]
@marionf, þakka þér kærlega!
